lib/charms/observability_libs/v1/kubernetes_service_patch.py

```diff
@@ -202,6 +222,7 @@ def __init__(
         self.framework.observe(charm.on.install, self._patch)
         self.framework.observe(charm.on.upgrade_charm, self._patch)
         self.framework.observe(charm.on.update_status, self._patch)
+        self.framework.observe(charm.on.remove, self._remove_service)
```

Comment:
Iiuc, we're not addressing the scenario of type change across an upgrade. I.e. a charm uses ClusterIP and then switches to LoadBalancer. Or vice versa.

Removing the service on stop would probably address this. Any reason to prefer remove over stop here?

Reply:
yeah this was just an oversight, thought it was the remove hook that comes before upgrade, thanks for noticing!